lib/navigation/preprocess_keyence.cpp

```diff
@@ -1,25 +1,31 @@
 #include "preprocess_keyence.hpp"
 
+#include <cstdint>
+
 namespace hyped::navigation {
 
-KeyencePreprocessor::KeyencePreprocessor()
+KeyencePreprocessor::KeyencePreprocessor(core::ILogger &logger)
+    : log_(logger),
+      previous_data_status_(KeyenceDataStatus::kAgreed)
 {
-  // TODOLater: implement
 }
-
-SensorChecks checkKeyenceAgrees(const core::KeyenceData &keyence_data)
+SensorChecks KeyencePreprocessor::checkKeyenceAgrees(const core::KeyenceData &keyence_data)
 {
-  /*
-   TODOLater: implement:
-   roughly:
-   - if keyence data disagrees and does_keyence_agree_ is false, fail state, return kUnacceptable
-   - if keyence data disagrees and does_keyence_agree_ is true, set does_keyence_agree_ false return
-   kAcceptable
-   - if keyence data agrees and does_keyence_agree_ is false, et does_keyence_agree_ true return
-   kAcceptable
-   - if keyence data agrees and does_keyence_agree_ is true, all is good in the world, return
-   kAcceptable.
-  */
+  KeyenceDataStatus current_data_status = KeyenceDataStatus::kDisagreed;
+  for (std::size_t i = 0; i < keyence_data.size() - 1; ++i) {
+    if (keyence_data.at(i) != keyence_data.at(i + 1)) {
```

Comment:
Why are we only comparing neighbours? What's going on here?

If the fundamental assumption is that there will only ever be two keyence readings then just compare those two directly?

Reply:
This is designed without assuming that there will only ever be 2 keyence readings. Therefore we set a disagreement flag if any keyence sensor disagrees with the others and check it on the next iteration (making comparison of neighbours a reasonable method surely?). 2 disagreements in a row is a critical failure.